mp/src/game/client/momentum/ui/HUD/hud_speedometer_label.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <vgui_controls/Label.h>
+#include "mom_shareddefs.h"
+
+namespace vgui
+{
+    class IScheme;
+}
+
+// for ranged based coloring
+struct Range_t
+{
+    int min;
+    int max;
+    Color color;
+};
+typedef CUtlVector<Range_t> RangeList;
+
+class SpeedometerLabel : public vgui::Label
+{
+    DECLARE_CLASS_SIMPLE(SpeedometerLabel, vgui::Label);
+
+  public:
+    SpeedometerLabel(Panel *parent, const char *panelName, SpeedometerColorize_t colorizeType);
+
+    void ApplySchemeSettings(vgui::IScheme *pScheme) override;
+    void OnThink() override; // for applying fadeout
+    void PerformLayout() override;
+    
+    void SetVisible(bool bVisible) override;
+
+    void Update(float value); // main function; called when updating the label
+    void Reset();
+    void SetText(int value);
+    void SetText(float value) { SetText(RoundFloatToInt(value)); }
+
+    // for separate comparison label
+    void SetCustomDiff(float diff);
+    // colorize will function as no color option if false
+    void SetDrawComparison(bool bEnabled) { m_bDrawComparison = bEnabled; }
+
+    // fadeout related functions
+    void SetFadeOutAnimation(char *animationName, float *animationAlpha);
+    void ClearFadeOutAnimation();
+    bool HasFadeOutAnimation() { return m_pSzAnimationName && m_pSzAnimationName[0] && m_pFlAlpha; }
+
+    // getter/setters
+    SpeedometerUnits_t GetUnitType() { return m_eUnitType; }
+    void SetUnitType(SpeedometerUnits_t type) { m_eUnitType = type; }
+    void SetUnitType(int type) { SetUnitType(SpeedometerUnits_t(type)); }
+
+    SpeedometerColorize_t GetColorizeType() { return m_eColorizeType; }
+    void SetColorizeType(SpeedometerColorize_t type) { m_eColorizeType = type; Reset(); }
+    void SetColorizeType(int type) { SetColorizeType(SpeedometerColorize_t(type)); }
```

Comment:
You should probably bounds check this `type` param with the enum, by making a `SPEEDOMETER_COLORIZE_FIRST` and `SPEEDOMETER_COLORIZE_LAST`. The [network packet reading stuff](https://github.com/momentum-mod/game/blob/develop/mp/src/game/shared/momentum/mom_ghostdefs.h#L359) did this recently, but it's generally good practice to do this when converting int -> enum.